lib/Document.php

```diff
@@ -180,23 +180,30 @@ public function createProperty(string $name, $value = null, ?array $parameters =
 
         $class = null;
 
+        // If a VALUE parameter is supplied, we have to use that
+        // According to https://datatracker.ietf.org/doc/html/rfc5545#section-3.2.20
+        //  If the property's value is the default value type, then this
+        //  parameter need not be specified.  However, if the property's
+        //  default value type is overridden by some other allowable value
+        //  type, then this parameter MUST be specified.
+        if (isset($parameters['VALUE'])) {
+            $class = $this->getClassNameForPropertyValue($parameters['VALUE']);
+        }
+
         if ($valueType) {
             // The valueType argument comes first to figure out the correct
             // class.
             $class = $this->getClassNameForPropertyValue($valueType);
         }
```

Comment:
If $valueType has been specified, and $parameters['VALUE'] has been set, then $class is calculated twice, the first result is immediately overwritten by the 2nd.

What am I not understanding?

Reply:
No, that is correct. But now that you phrase it like that I might actually change it so that when `$valueType` evaluates to false and `$parameters['VALUE']` is set I will set it to `$parameters['VALUE']` instead. Then it would be truish in line 193 and the calculation only happens once.

Would that be a better approach?

Reply:
> Would that be a better approach?

Yes, something like having an `else if` on the end of `if ($valueType) {`

Reply:
The code you have done also works.